lib/castle/client.rb

```diff
@@ -25,7 +25,7 @@ def authenticate(options = {})
         command = Castle::Commands::Authenticate.new(@context).build(options || {})
         begin
           @api.request(command).merge('failover' => false, 'failover_reason' => nil)
-        rescue Castle::RequestError => error
+        rescue Castle::RequestError, Castle::InternalServerError => error
```

Comment:
@nijikon do you know in which scenario `Castle::RequestError` is still raised?

Reply:
Of course, https://github.com/castle/castle-ruby/blob/master/lib/castle/api.rb#L44